pydmd/bopdmd.py

```diff
@@ -283,6 +292,147 @@ def _varpro_opts_warn(self):
                 )
                 warnings.warn(msg.format(opt_name, opt_value, opt_max))
 
+    @staticmethod
+    def _diff_func(eigenvalues, omega, ind, absolute_diff):
+        """Create an array of differences for the imaginary component.
+
+        Used to find the eigenvalue pairs that most closely approximate
+        conjugate pairs. There are two modes of operation for this function
+        given by the `absolute_diff` variable.
+
+        `absolute_diff == True` should be used when there is not an equal number
+        of eigenvalues with positive and negative imaginary components. The
+        difference array is simply found by looking at the absolute
+        difference between imaginary components.
+
+        `absolute_diff == False` should be used when there is an equal number of
+        eigenvalues with positive and negative imaginary components. Here we can
+        restrict ourselves to only looking at the differences between the input
+        eigenvalue (`omega`) and the eigenvalues with an oppositely signed
+        imaginary component. Also used to catch an edge case for `omega.imag
+        == 0`.
+
+        :param eigenvalues: Vector of eigenvalues to calculate the difference
+        :param omega: Currently considered eigenvalue
+        :param ind: Index of omega inside of eigenvalues vector
+        :param absolute_diff: Flag dictating difference matrix calculation.
+        :return: Difference vector between omega and other eigenvalues.
+        """
+        if absolute_diff:
+            diff = np.abs(np.abs(eigenvalues.imag) - np.abs(omega.imag))
+            diff[ind] = np.nan
+        else:
+            sign = np.sign(omega.imag)
+            # Catch the edge case of the eigenvalues being exactly 0.
+            if sign == 0.0:
+                diff = np.abs(np.abs(eigenvalues.imag) - omega.imag)
+                diff[ind] = np.nan
+            else:
+                same_sign_index = np.sign(eigenvalues.imag) == sign
+                opp_sign_eigs = np.copy(eigenvalues.imag)
+                opp_sign_eigs[same_sign_index] = np.nan
+                diff = np.abs(np.abs(opp_sign_eigs) - np.abs(omega.imag))
+
+        return diff
```

Comment:
I propose an early return here:
```suggestion
            diff[ind] = np.nan
            return diff
 
        sign = np.sign(omega.imag)
        
        # Catch the edge case of the eigenvalues being exactly 0.
        if sign == 0.0:
            diff = np.abs(np.abs(eigenvalues.imag) - omega.imag)
            return diff[ind] = np.nan
            
        same_sign_index = np.sign(eigenvalues.imag) == sign
        opp_sign_eigs = np.copy(eigenvalues.imag)
        opp_sign_eigs[same_sign_index] = np.nan
        return np.abs(np.abs(opp_sign_eigs) - np.abs(omega.imag))
```